### PR TITLE
[Task][66490] Handle Event UI for Home Screen

### PR DESCRIPTION
**Pull request by Lomahs1401**
## Related Tickets
- [#Task 66490: Handle Event UI for Home Screen](https://edu-redmine.sun-asterisk.vn/issues/66490)
## WHAT
 - Click on the image item to show recipe name
 - Enter input in search bar to fetch data
 - Click category button to fetch data
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/FoodRecipeApp/assets/77973385/6c1f69d8-ff2e-48a6-adce-420f3ae22d61

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 import android.view.LayoutInflater
 import android.view.ViewGroup
+import androidx.core.content.ContextCompat
 import androidx.recyclerview.widget.RecyclerView
+import com.example.foodrecipeapp.R
 import com.example.foodrecipeapp.databinding.ItemCategoryBinding
+import com.example.foodrecipeapp.screen.home.HomePresenter
 
-class CategoryAdapter : RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder>() {
+class CategoryAdapter(private val context: Context, private val presenter: HomePresenter) :
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt]: không truyền context vô e

File: app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt [hunk 2]
```diff
@@ -29,9 +37,37 @@ class CategoryAdapter : RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder>
...
+    }
+
+    inner class CategoryViewHolder(private val binding: ItemCategoryBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bindData(category: String, position: Int) {
             binding.btnCategory.text = category
+
+            if (position == currentIndex) {
+                binding.btnCategory.typeface = Typeface.DEFAULT_BOLD
+                binding.btnCategory.setBackgroundColor(ContextCompat.getColor(context, R.color.colorAccent))
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt]:
```suggestion
                binding.btnCategory.typeface = Typeface.DEFAULT_BOLD
                binding.btnCategory.setBackgroundColor(ContextCompat.getColor(binding.root.context, R.color.colorAccent))
```

File: app/src/main/java/com/example/foodrecipeapp/adapter/HomeChildAdapter.kt [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
 import com.example.foodrecipeapp.databinding.LayoutRecentRecipeHomeChildBinding
 import com.example.foodrecipeapp.databinding.LayoutSearchHomeChildBinding
+import com.example.foodrecipeapp.listener.OnRecipeItemClickListener
+import com.example.foodrecipeapp.screen.home.HomePresenter
 import com.example.foodrecipeapp.utils.base.BaseViewHolder
 
-class HomeChildAdapter(private val resources: Resources) :
-    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class HomeChildAdapter(
+    private val context: Context,
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/adapter/HomeChildAdapter.kt]: bỏ context luôn e

File: app/src/main/java/com/example/foodrecipeapp/screen/MainActivity.kt [hunk 4]
```diff
@@ -14,6 +14,8 @@ import com.example.foodrecipeapp.utils.base.BaseViewBindingActivity
 
 class MainActivity : BaseViewBindingActivity<ActivityMainBinding>() {
 
+    private lateinit var pagerAdapter: MainPageAdapter
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/screen/MainActivity.kt]: dùng lazy

File: app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt [hunk 5]
```diff
@@ -12,16 +12,27 @@ import com.example.foodrecipeapp.data.repo.RecipeRepo
...
     private lateinit var homePresenter: HomePresenter
+    private val recipeItemClickListener: OnRecipeItemClickListener = this
 
     private val homeChildAdapter: HomeChildAdapter by lazy {
-        HomeChildAdapter(context = context, resources = resources)
+        HomeChildAdapter(
+            context = requireContext(),
+            resources = resources,
+            presenter = homePresenter,
+            recipeItemClickListener = recipeItemClickListener
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt]:
```suggestion
            presenter = homePresenter,
            recipeItemClickListener = this
```

File: app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt [hunk 6]
```diff
@@ -12,16 +12,27 @@ import com.example.foodrecipeapp.data.repo.RecipeRepo
...
 
 class HomeFragment :
     BaseViewBindingFragment<FragmentHomeBinding>(),
     HomeContract.View,
-    OnItemRecyclerViewClickListener<Recipe> {
+    OnItemRecyclerViewClickListener<Recipe>,
+    OnRecipeItemClickListener {
+
     private lateinit var homePresenter: HomePresenter
+    private val recipeItemClickListener: OnRecipeItemClickListener = this
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt]: hơi dư e

File: app/src/main/java/com/example/foodrecipeapp/utils/ext/FragmentExt.kt [hunk 7]
```diff
@@ -15,13 +15,14 @@ fun Fragment.addFragment(
...
             }
+
             setCustomAnimations(
                 R.anim.slide_in_right,
                 R.anim.slide_in_right,
                 R.anim.slide_out_right,
                 R.anim.slide_out_right
             )
-            add(containerId, fragment, tag)
+            replace(containerId, fragment, tag)
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/utils/ext/FragmentExt.kt]: sao lại làm như này e, mình có 2 hàm 1 add, 1 replace chứ nhỉ

File: app/src/main/res/drawable/btn_category_normal.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="@android:color/white" />
+    <corners android:radius="@dimen/dp_40" />
+</shape>
```
Comment (ghost) [inline app/src/main/res/drawable/btn_category_normal.xml]: thêm dòng trống

File: app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt (hunk 1 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt]: Done

File: app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt (hunk 2 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/adapter/CategoryAdapter.kt]: Done

File: app/src/main/java/com/example/foodrecipeapp/adapter/HomeChildAdapter.kt (hunk 3 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/adapter/HomeChildAdapter.kt]: Done

File: app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt (hunk 5 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt]: Done

File: app/src/main/res/drawable/btn_category_pressed.xml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="@color/colorAccent" />
+    <corners android:radius="@dimen/dp_40" />
+</shape>
```
Comment (ghost) [inline app/src/main/res/drawable/btn_category_pressed.xml]: .

File: app/src/main/res/drawable/btn_category_selector.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+<!--        android:drawable="@drawable/btn_category_pressed" />-->
+<!--    <item-->
+<!--        android:state_focused="true"-->
+<!--        android:drawable="@drawable/btn_category_pressed" />-->
+    <item
+        android:state_selected="true"
+        android:drawable="@drawable/btn_category_pressed" />
+    <item
+        android:drawable="@drawable/btn_category_normal" />
+</selector>
```
Comment (ghost) [inline app/src/main/res/drawable/btn_category_selector.xml]: .

File: app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt (hunk 6 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt]: Done

File: app/src/main/java/com/example/foodrecipeapp/utils/ext/FragmentExt.kt (hunk 7 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/utils/ext/FragmentExt.kt]: Done

File: app/src/main/res/drawable/btn_category_normal.xml (hunk 8 above)
Comment (Lomahs1401) [reply app/src/main/res/drawable/btn_category_normal.xml]: Không đẩy lên nữa

File: app/src/main/res/drawable/btn_category_selector.xml (hunk 10 above)
Comment (Lomahs1401) [reply app/src/main/res/drawable/btn_category_selector.xml]: Không đẩy lên nữa

File: app/src/main/res/drawable/btn_category_pressed.xml (hunk 9 above)
Comment (Lomahs1401) [reply app/src/main/res/drawable/btn_category_pressed.xml]: Không đẩy lên nữa

File: app/src/main/java/com/example/foodrecipeapp/screen/MainActivity.kt (hunk 4 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/screen/MainActivity.kt]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**